src/debug/telemetry/telemetry2_slot.h

```diff
@@ -0,0 +1,72 @@
+/* SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) 2024 Intel Corporation
+ */
+
+#ifndef _ZEPHYR_SOC_INTEL_ADSP_TELEMETRY2_SLOT
+#define _ZEPHYR_SOC_INTEL_ADSP_TELEMETRY2_SLOT
+
+/*
+ * Telemetry2 slot is taken from SRAM window divided into several
+ * chunks with simple header declared bellow.
+ *
+ * "type" describes the contents of the chunk
+ * "size" describes the size of the chunk in bytes
+ *
+ * After the chunk another chunk is expected to follow, if there is
+ * enough space for another header. If type is 0 and size 0 it
+ * indicated that the rest of the slot is unused, and this header is
+ * free for taking.
+ *
+ * Each chunk is aligned to start at 64-byte cache line boundary.
+ *
+ * For example:
+ *      --------------------------------------------------  ---
+ *      | magic = TELEMETRY2_PAYLOAD_MAGIC               |   |
+ *      | hdr_size = 64                                  |   |
+ *      | total_size = 320                               |   |
+ *      | abi = TELEMETRY2_PAYLOAD_V0_0                  |  64 bytes
+ *      | tstamp = <aligned with host epoch>             |   |
+ *      | <padding>                                      |   |
+ *	--------------------------------------------------  ---
+ *	| type = ADSP_DW_TELEMETRY2_ID_THREAD_INFO       |   |
+ *      | size = 256                                     |  256 bytes
+ *	|    chunk data                                  |   |
+ *	|    ...                                         |   |
+ *	--------------------------------------------------  ---
+ *	| type = ADSP_DW_TELEMETRY2_TYPE_EMPTY           |
+ *      | size = 0                                       |
+ *	--------------------------------------------------
+ *
+ * The above depicts a telemetry2 slot with thread analyzer data
+ * of 256 bytes (including the header) in the first chunk and
+ * the rest of the slot being free.
+ */
+
+#include <stdint.h>
+
+#define TELEMETRY2_PAYLOAD_MAGIC 0x1ED15EED
+
+#define TELEMETRY2_PAYLOAD_V0_0 0
+
+/* Telemetry2 pauload hdr */
+struct telemetry2_payload_hdr {
+	uint32_t magic;		// used to identify valid data
+	uint32_t hdr_size;	// size of this header only in bytes
+	uint32_t total_size;	// total size of the whole payload in bytes
+	uint32_t abi;		// ABI version, can be used by tool to warn users if too old.
+	uint64_t tstamp;	// aligned with host epoch.
```

Comment:
how do we set the epoch?

Reply:
No, for the moment we don't. Its from Liam's comment. I actually do not think we should have it here. In the memory window each telemetry2 client is updating just their own chunk, and there can not be any universal timestamp for all of them. Each should have their own time-stamps indicating the time when the chunk was last updated. I would drop it. What do you say @lgirdwood ?

Reply:
Yes, but in the case of debug window, the chunks may be updated independently. There is no universal timestamp that would be valid for the whole payload. If a timestamp is needed, each chunk should have its own stamp.